assets/activation-scripts/activate.d/generate-tcsh-startup-commands.bash

```diff
@@ -35,13 +35,15 @@ generate_tcsh_startup_commands() {
     # Restore environment variables set in the previous bash initialization.
     $_sed -e 's/^/unsetenv /' -e 's/$/;/' "$_FLOX_ACTIVATION_STATE_DIR/del.env"
     $_sed -e 's/^/setenv /' -e 's/=/ /' -e 's/$/;/' "$_FLOX_ACTIVATION_STATE_DIR/add.env"
-
-    # Propagate $_activate_d to the environment.
-    echo "setenv _activate_d '$_activate_d';"
-    # Propagate $_flox_activate_tracer to the environment.
-    echo "setenv _flox_activate_tracer '$_flox_activate_tracer';"
   fi
 
+  # Propagate $_activate_d to the environment.
+  echo "setenv _activate_d '$_activate_d';"
+  # Propagate $_flox_activate_tracer to the environment.
+  echo "setenv _flox_activate_tracer '$_flox_activate_tracer';"
+  # Propagate $_flox_env_helper to the environment.
+  echo "setenv _flox_env_helper '$_flox_env_helper';"
+
```

Comment:
followup: I believe we've agreed that we want to move away from passing variables with environment variables. Not completely sure but I think exporting these vars was introduced since we made that decision. The fact that we then discovered a bug with the approach seems to confirm that passing via environment variables is an unreliable approach, and I'd prefer changing the approach rather than patching a faulty approach